src/leveled_eval.erl

```diff
@@ -0,0 +1,560 @@
+%% -------- Eval Functions ---------
+%%
+%% Support for different eval expressions within leveled 
+%%
+
+-module(leveled_eval).
+
+-export([apply_eval/4, generate_eval_expression/2, generate_eval_function/2]).
+
+%%%============================================================================
+%%% External API
+%%%============================================================================
+
+generate_eval_function(EvalString, Substitutions) ->
+    {ok, ParsedEval} = generate_eval_expression(EvalString, Substitutions),
+    fun(Term, Key) ->
+        apply_eval(ParsedEval, Term, Key, maps:new())
+    end.
+
+generate_eval_expression(EvalString, Substitutions) ->
+    {ok, Tokens, _EndLine} = leveled_evallexer:string(EvalString),
+    case leveled_filter:substitute_items(Tokens, Substitutions, []) of
+        {error, Error} ->
+            {error, Error};
+        UpdTokens ->
+            leveled_evalparser:parse(UpdTokens)
+    end.
+
+
+apply_eval({eval, Eval}, Term, Key, AttrMap) ->
+    apply_eval(Eval, Term, Key, AttrMap);
+apply_eval({'PIPE', Eval1, 'INTO', Eval2}, Term, Key, AttrMap) ->
+    apply_eval(Eval2, Term, Key, apply_eval(Eval1, Term, Key, AttrMap));
+apply_eval({
+        delim, {identifier, _, InKey}, {string, _, Delim}, ExpKeys},
+        Term, Key, AttrMap) ->
+    TermToSplit = term_to_process(InKey, Term, Key, AttrMap),
+    CptTerms = string:split(TermToSplit, Delim, all),
+    L = min(length(CptTerms), length(ExpKeys)),
```

Comment:
in OTP 26 there is now a `How` parameter for zip. Using `trim` would reduce this code. But only introduced for OTP26.